gcc/config/riscv/arcv.cc

```diff
@@ -556,6 +556,45 @@ arcv_sched_init (void)
   sched_state.last_scheduled_insn = 0;
 }
 
+/* Return the next possible fusible insn.  */
+
+static rtx_insn *
```

Comment:
Does it make sense to refactor like this ?
```suggestion
static rtx_insn *
arcv_next_fusible_insn (rtx_insn *insn)
{
  while (insn)
    {
      insn = NEXT_INSN (insn);

      if (insn == 0)
        break;

      if (DEBUG_INSN_P (insn)
          || NOTE_P (insn))
        continue;

      if (NOTE_INSN_BASIC_BLOCK_P (insn))
        return NULL;

      if (GET_CODE (insn) == CODE_LABEL
          || GET_CODE (insn) == BARRIER
          || GET_CODE (PATTERN (insn)) == USE)
        continue;

      if (JUMP_TABLE_DATA_P (insn))
        return NULL;

      break;
    }

  return insn;
}
```